### DISCUSSION TITLE
What API documentation syntax that `protocol-http` use?

### DISCUSSION THREAD

**Discussion by postmodern** [socketry/protocol-http#55] [Q&A] [answered]
I'm trying to view the API documentation for [protocol-http](https://rubydoc.info/gems/protocol-http/Protocol/HTTP/Request) on https://rubydoc.info which uses YARD, but it appears that `protocol-http` uses some other kind of `@tag` based documentation syntax I haven't seen before? This results in some malformed API documentation that's difficult to read.

**Comment by ioquatix** [comment, marked as answer]
The best place to read the documentation is on the GitHub pages for the project, as linked from the readme: https://socketry.github.io/protocol-http/index.html

The actual syntax is implemented by https://github.com/ioquatix/decode - it's very similar to YARD but has a more consistent signature for programatically extracting types.
